### PR TITLE
[ARM][Thumb2] Mark BTI-clearing instructions as scheduling region boundaries

### PR DESCRIPTION
**Pull request by vhscampos**
Following https://github.com/llvm/llvm-project/pull/68313 this patch extends the idea to M-profile PACBTI.

The Machine Scheduler can reorder instructions within a scheduling region depending on the scheduling policy set. If a BTI-clearing instruction happens to partake in one such region, it might be moved around, therefore ending up where it shouldn't.

The solution is to mark all BTI-clearing instructions as scheduling region boundaries. This essentially means that they must not be part of any scheduling region, and as consequence never get moved:

 - PAC
 - PACBTI
 - BTI
 - SG

Note that PAC isn't BTI-clearing, but it's replaced by PACBTI late in the compilation pipeline.

As far as I know, currently it isn't possible to organically obtain code that's susceptible to the bug:

- Instructions that write to SP are region boundaries. PAC seems to always be followed by the pushing of r12 to the stack, so essentially PAC is always by itself in a scheduling region.
- CALL_BTI is expanded into a machine instruction bundle. Bundles are unpacked only after the last machine scheduler run. Thus setjmp and BTI can be separated only if someone deliberately run the scheduler once more.
- The BTI insertion pass is run late in the pipeline, only after the last machine scheduling has run. So once again it can be reordered only if someone deliberately runs the scheduler again.

Nevertheless, one can reasonably argue that we should prevent the bug in spite of the compiler not being able to produce the required conditions for it. If things change, the compiler will be robust against this issue.

The tests written for this are contrived: bogus MIR instructions have been added adjacent to the BTI-clearing instructions in order to have them inside non-trivial scheduling regions.

### CODE REVIEW

**Comment by vhscampos** [comment]
Since the previous PR had already been approved, I plan to merge this one on Monday if no objections arise.

File: llvm/test/CodeGen/ARM/misched-branch-targets.mir [hunk 1]
```diff
@@ -0,0 +1,169 @@
+# REQUIRES: asserts
```
Comment (davemgreen) [inline llvm/test/CodeGen/ARM/misched-branch-targets.mir:1]: It's customary to put REQUIRES after the RUN lines, but either should work.

**Review by davemgreen — APPROVED** [review]
The changes LGTM, compared to the old version.

File: llvm/test/CodeGen/ARM/misched-branch-targets.mir (hunk 1 above)
Comment (vhscampos) [reply llvm/test/CodeGen/ARM/misched-branch-targets.mir]: Fixed. Thanks

**Pull request merged**